pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -351,3 +340,19 @@ func hashFile(path string) (string, error) {
 
 	return hex.EncodeToString(hasher.Sum(nil)), nil
 }
+
+func getWhitelistedEnvVars() map[string][]byte {
```

Comment:
Is there a specific reason we are duplicating logic already found in the codebase i.e. here: https://github.com/rancher/rancher/blob/d657043de1dc2110562d9035b6c1c6dc9d9232ed/pkg/jailer/jailer.go#L80 ?

Reply:
The functions operate on different data structures which is why I decided to copy it and refactor the copy. The `WhitelistEnvvars` function returns a list of string in which each item is a string formatted with `name=value`. The copy I made returns a `map[string][]byte` which is a better fit for the secret used in RKE2.

`WhitelistEnvvars` also takes a parameter which is a seed of already existing `name=value` strings coming from the `*exec.Cmd` object. This parameter is not needed in the RKE2 case. I could have re-used that function and added some extra code to parse a list of `name=value` strings into a `map[string][]byte` but it felt inefficient and ugly to me.

Reply:
In this case the signature of the function you introduced is a lot more natural and can be used instead of the one @Oats87 referenced and have jailer use the function you introduced to range through the returned map and append `name=value` pairs to the slice.
The only question here is where this new function should reside, `setting.go` is one option I guess.

I don't feel too strongly about having to do this, though a perfectionist in me would prefer for it to be done.

Reply:
I think part of the problem is that [Cmd.Env](https://pkg.go.dev/os/exec#Cmd) which is part of the standard library is a list of `name=value` strings. If we want to seed the map of environment variables with initial data from this object, as done in the RKE1 code, we'll have to parse it.

It might be possible to change that functions' signature to use a callback function instead such as,

```go
settings.IterWhitelistedEnvVars(func (name, value string) {
  cmd.Env = append(cmd.Env, name + "=" + value)
})

// ...

settings.IterWhitelistedEnvVars(func (name, value string) {
  secret.Data[name] = []byte(value)
})
```